src/components/connect.js

```diff
@@ -223,20 +223,20 @@ function connect(mapPropsToRequestsToProps, defaults, options) {
 
       refetchDataFromMappings(mappings) {
         mappings = coerceMappings(mappings)
-        Object.keys(mappings).forEach(prop => {
+        return Promise.all(Object.keys(mappings).map(prop => {
           const mapping = mappings[prop]
 
           if (Function.prototype.isPrototypeOf(mapping)) {
             this.setAtomicState(prop, new Date(), mapping, (...args) => {
```

Comment:
doesn't this need a `return` and inside of `setAtomicState` as well? (otherwise the `return` below isn't doing anything)